File: 0567-permutation-in-string/0567-permutation-in-string.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        int n = s1.length();
        int m = s2.length();

        int[] f1 = new int[26];
        int[] f2 = new int[26];

        for(int i=0;i<n;i++){
            f1[s1.charAt(i) -'a']++;
        }

        int i =0,j=0;
        while(j < m){
            f2[s2.charAt(j) - 'a']++;
            if(j-i+1 > n){
                f2[s2.charAt(i) -'a']--;
                i++;
            }
            if(Arrays.equals(f1,f2)){
                return true;
            }
            j++;
        }
        return false;
    }
}

File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        int n = s1.size();
        int m = s2.size();

        vector<int> f1(26, 0);
        vector<int> f2(26, 0);

        for(int i=0;i<n;i++){
            f1[s1[i] -'a']++;
        }

        int i =0,j=0;
        while(j < m){
            f2[s2[j] - 'a']++;
            if(j-i+1 > n){
                f2[s2[i] -'a']--;
                i++;
            }
            if(f1 == f2){
                return true;
            }
            j++;
        }
        return false;
    }
};